python/mxnet/text/glossary.py

```diff
@@ -0,0 +1,654 @@
+        download_file_path = os.path.join(embed_dir, download_file)
+
+        embed_cls = Embedding.embed_registry[embed_name]
+        expected_file_hash = embed_cls.pretrain_file_sha1[pretrain_file]
+
+        if hasattr(embed_cls, 'pretrain_archive_sha1'):
+            expected_download_hash = \
+                embed_cls.pretrain_archive_sha1[download_file]
+        else:
+            expected_download_hash = expected_file_hash
+
+        # The specified pretrained embedding file does not exist or fails to
+        # match its expected SHA-1 hash.
+        if not os.path.isfile(pretrain_file_path) or \
+                not check_sha1(pretrain_file_path, expected_file_hash):
+            # If download_file_path exists and matches
+            # expected_download_hash, there is no need to download.
+            download(url, download_file_path,
+                     sha1_hash=expected_download_hash)
+
+        # If the downloaded file does not match its expected SHA-1 hash,
+        # we do not encourage to load embeddings from it in case that its
+        # data format is changed.
+        assert check_sha1(download_file_path, expected_download_hash), \
+            'The downloaded file %s does not match its expected SHA-1 ' \
+            'hash. This is caused by the changes at the externally ' \
+            'hosted pretrained embedding file(s). Since its data format ' \
+            'may also be changed, it is discouraged to continue to use ' \
+            'mxnet.text.Embedding.create_embedding(%s, **kwargs) to load ' \
+            'the pretrained embedding %s. If you still wish to load the ' \
+            'changed embedding file, please specify its path %s via ' \
+            'pretrain_file of mxnet.text.Embedding(). It will be loaded ' \
+            'only if its data format remains unchanged.' % \
+            (download_file_path, embed_name, embed_name, download_file_path)
+
+        ext = os.path.splitext(download_file)[1]
+        if ext == '.zip':
+            with zipfile.ZipFile(download_file_path, 'r') as zf:
+                zf.extractall(embed_dir)
+        elif ext == '.gz':
+            with tarfile.open(download_file_path, 'r:gz') as tar:
+                tar.extractall(path=embed_dir)
+        return pretrain_file_path
+
+    def _load_embedding(self, pretrain_file_path, token_delim):
+        """Load embedding vectors from the pre-trained embedding file.
+
+        The largest valid index of idx_to_vec maps to the initialized embedding
+        vector for every special token, such as an unknown token and a padding
+        token.
+        """
+        with open(pretrain_file_path, 'r', encoding='utf8') as f:
+            lines = f.readlines()
+
+        logging.info('Loading pretrained embedding vectors from %s'
+                     % pretrain_file_path)
+
+        vec_len = None
+        all_elems = []
+        idx_to_token = []
+        for line in tqdm(lines, total=len(lines)):
```

Comment:
Let’s avoid adding dependencies such as tqdm.

Reply:
Thanks. Since https://github.com/apache/incubator-mxnet/blob/master/example/gluon/tree_lstm/dataset.py uses tqdm, I assume that tqdm dependency is added. Let me know if you still prefer removing such dependency.

Reply:
Resolved. (tqdm is removed)